Realm/RLMSyncUser.mm

```diff
@@ -252,7 +252,7 @@ - (RLMRealm *)permissionRealmWithError:(NSError **)error {
 }
 
 - (NSURL *)authenticationServer {
-    if (!_user) {
+    if (!_user || _user->token_type() == SyncUser::TokenType::Admin) {
```

Comment:
My original reasoning was that the code that method called into took care of returning null or nil or whatever for admin token users, but that doesn't appear to be true. (It might have been true before the object store changes this is trying to incorporate.)